akka-http-core/src/test/scala/akka/http/scaladsl/util/FastFutureSpec.scala

```diff
@@ -176,8 +177,12 @@ class FastFutureSpec extends FreeSpec with Matchers {
     testLazily()
   }
 
-  def failF: PartialFunction[Any, Nothing] = PartialFunction(_ ⇒ throw TheException)
```

Comment:
These looks suspiciously like there was a reason they were made like this, but I don't know it. Was the old one more performant/less wasteful as it doesn't really do any matching perhaps?

Reply:
It's deprecated in Scala 2.12, hence the change.

Reply:
https://www.scala-lang.org/api/current/scala/PartialFunction$.html#apply[A,B](f:A=%3EB):PartialFunction[A,B]

> (Since version 2.12.5) For converting an ordinary function f to a partial function pf, use val pf: PartialFunction[A, B] = { case x => f(x) }. For creating a new PartialFunction, use an explicit type annotation instead, like in val pf: PartialFunction[Int, String] = { case 1 => "one" }.

Reply:
👍 thanks for the rationale

Reply:
I also learned it during this warning fix :)